proto/cosmos/authz/v1beta1/authz.proto

```diff
@@ -35,3 +35,8 @@ message AuthorizationGrant {
   google.protobuf.Any       authorization = 1 [(cosmos_proto.accepts_interface) = "Authorization"];
   google.protobuf.Timestamp expiration    = 2 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
 }
+
+// VoteAuthorization allows the grantee to vote on a proposal on behalf of the granter's account.
+message VoteAuthorization {
+  option (cosmos_proto.implements_interface) = "Authorization";
```

Comment:
How about authorizing for specific type of proposals? That would be more granular I believe.
ProposalType: Any

It can have one of  `All`, `TextProposal`, `SoftwareUpgradeProposal`, `ParamChangeProposal` etc.

@amaurymartiny wdyt?

Reply:
I agree 👍

Reply:
I didn't get how we can achieve this, `Accept` method doesn't have access to the state/keeper.

Reply:
Ah yes, we should find a cleaner way to handle this

Reply:
So we can use GenericAuthorization if there is nothing special to check here. Adding state access seems out of scope and feels like we should address with the ADR 033 approach if we do...